docs2/site/docs/getting-started/errors.md

```diff
@@ -200,3 +200,36 @@ For example, to show the stack traces for unhandled errors during development, y
 You can also write your own implementation of `IErrorInfoProvider`. For instance, you might want to override
 the numerical codes provided by GraphQL.NET for validation errors, or reveal stack traces
 only to logged-in administrators.
+
+## <a name="ValidationErrors"></a>Validation error reference list
+
+Here is a full list of validation errors produced by GraphQL.NET:
+
+Rule                         | Code                             | Number
+-----------------------------|----------------------------------|-----------
+UniqueOperationNames         | UNIQUE_OPERATION_NAMES           | 5.2.1.1
+LoneAnonymousOperation       | LONE_ANONYMOUS_OPERATION         | 5.2.2.1
+SingleRootFieldSubscriptions | SINGLE_ROOT_FIELD_SUBSCRIPTIONS  | 5.2.3.1
+FieldsOnCorrectType          | FIELDS_ON_CORRECT_TYPE           | 5.3.1
+OverlappingFieldsCanBeMerged | OVERLAPPING_FIELDS_CAN_BE_MERGED | 5.3.2
+ScalarLeafs                  | SCALAR_LEAFS                     | 5.3.3
+KnownArgumentNames           | KNOWN_ARGUMENT_NAMES             | 5.4.1
+UniqueArgumentNames          | UNIQUE_ARGUMENT_NAMES            | 5.4.2
+ProvidedNonNullArguments     | PROVIDED_NON_NULL_ARGUMENTS      | 5.4.2.1
+UniqueFragmentNames          | UNIQUE_FRAGMENT_NAMES            | 5.5.1.1
+KnownTypeNames               | KNOWN_TYPE_NAMES                 | 5.5.1.2
+FragmentsOnCompositeTypes    | FRAGMENTS_ON_COMPOSITE_TYPES     | 5.5.1.3
+KnownFragmentNames           | KNOWN_FRAGMENT_NAMES             | 5.5.1.4
+NoUnusedFragments            | NO_UNUSED_FRAGMENTS              | 5.5.1.4
```

Comment:
![изображение](https://user-images.githubusercontent.com/21261007/91450506-0b885580-e885-11ea-88c7-a04b0d105ff4.png)

Reply:
I'd appreciate it if you'd verify the numbers that I listed.

Reply:
Maybe known fragment names should be 5.5.1.1 ?

Reply:
(because of "fragments must be a set of one.")

Reply:
UniqueFragmentNames is 5.5.1.1

Reply:
Yes, I know.  I don't know what you want me to do.

Reply:
I think KnownFragmentNames is 5.5.1.2 and KnownTypeNames is some sort of aggregated check which is not directly listed in the specification. KnownTypeNames ia about all types defined in document - variable definitions and fragment conditions. All 5.5 paragraph is about fragments. Variable definition is handled by 5.8.2 - `IsInputType` to some extent implies that the variable type must be defined.  Fragment type existence is handled by 5.5.1.2, so....  Maybe KnownTypeNames rule has become redundant and needs to be removed so as not to get confused?

Reply:
If we remove that check, then `use_undefined_fragment` incorrectly returns data.

> 5.5.1.1
> For each fragment definition fragment in the document
> Let fragmentName be the name of fragment.
> Let fragments be all fragment definitions in the document named fragmentName.
> **fragments must be a set of one.**

Fragments would be a set of zero, instead of one.  Test does not meet 5.5.1.1

Reply:
For reference, `use_undefined_fragment` is this:
```gql
query someDroids {
  r2d2: droid(id: ""3"") {
    ...unknown_fragment
    name
  }
}
```

Reply:
KnownFragmentNames has been updated to 5.5.1.1

Reply:
Actually that's wrong...... 5.5.1.1 is checking for duplicate fragments......

Reply:
Ok it needs to be 5.5.1.2 - `use_undefined_fragment` fails because it's not defined in the schema

Reply:
Oh I misread what you said.  let me check KnownTypeNames

Reply:
Ok I did it wrong again.  5.5.1.2 is "fragment spread TYPE EXISTENCE" and 5.5.2.1 is "Fragment spread TARGET DEFINED"

So I think KnownFragmentNames is 5.5.2.1 and KnownTypeNames is 5.5.1.2

Reply:
It seems to me that all this numbering will still need to be checked again someday. We will need to add the missing rules, then it will be possible to find out the exact numbering. I'm OK with any current numbering.